docs/release_note_guidelines.rst

```diff
@@ -0,0 +1,179 @@
+Release Note Guidelines
+=======================
+The guidelines below describe how to ensure consistency and clarity in release notes.
+
+Release notes are the first product documentation a user will see when updating the library. Therefore, we must take care to ensure the quality of release notes.
+
+For every pull request that changes how a user interacts with the library, a pull request should include a release note entry.
+
+
+Does the change require a release note?
++++++++++++++++++++++++++++++++++++++++
+
+A release note is required if a PR is user-facing, or if it meets any of the following conditions:
```

Comment:
```suggestion
A release note is required if a PR is user-impacting, or if it meets any of the following conditions:
```

or maybe `user-affecting`? Not sure the best option here but `facing` might not fully capture what we mean